src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsrestrictedpatientsapi/gateways/PrisonApiGateway.kt

```diff
@@ -34,6 +35,13 @@ class PrisonApiGateway(private val prisonApiWithAuthWebClient: WebClient) {
     .bodyToMono(object : ParameterizedTypeReference<List<Agency>>() {})
     .block()!!
 
+  fun getAgencyById(agencyId: String): Agency? = prisonApiWithAuthWebClient
+    .get()
+    .uri("/agencies/$agencyId")
```

Comment:
worth looking at the swagger docs - I think something like
https://api-dev.prison.service.justice.gov.uk/api/agencies/MDI?activeOnly=true&agencyType=INST&withAddresses=false&skipFormatLocation=true
would be better and then you won't need to check the agency type etc.

Reply:
Ah, yes that's a bit cleaner. Will sort that on the next PR